napalm/ios.py

```diff
@@ -0,0 +1,435 @@
+# Copyright 2015 Spotify AB. All rights reserved.
+#
+# The contents of this file are licensed under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with the
+# License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+from netmiko import ConnectHandler
+from base import NetworkDriver
+
+import re
+
+class IOSDriver(NetworkDriver):
+
+    def __init__(self, hostname, username, password, timeout=60):
+        self.hostname = hostname
+        self.username = username
+        self.password = password
+        self.timeout = timeout
+        self.device = None
+
+    def open(self):
+        """Opens a connection to the device."""
+        self.device = ConnectHandler(
+            device_type='cisco_ios',
+            ip=self.hostname,
+            username=self.username,
+            password=self.password)
+
+    def close(self):
+        """Closes the connection to the device."""
+        self.device.disconnect() 
+
+    def discard_config(self):
```

Comment:
Quick question about this method. Are you trying to revert changes after they have been committed? If yes, you should do that with the rollback method. Discard should just discard the candidate configuration before commit. For example:

    device.open()
    device.load_merge_configuration(filename='my_changes.txt')
    # I changed my mind and I don't want to commit:
    device.discard_config()
    
    # Let's try this other instead
    device.load_merge_configuration(filename='my_other_changes.txt')
    device.commit()
    
    # Oh shit! I broke it : )
    device.rollback()
    device.close()